src/insets.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2019 The kurbo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A description of the distances between the edges of two rectangles.
+
+use std::ops::Add;
+
+use crate::Rect;
+
+/// Insets from the edges of a rectangle.
+///
+///
+/// The inset value for each edge can be thought of as a delta computed from
+/// the center of the rect to that edge. For instance, with an inset of `2.0` on
+/// the x-axis, a rectange at `(0.0, 0.0)` with that inset added will be at
+/// `(-2.0, 0.0)`
+///
+/// Put alternatively, a positive inset represents increased distance from center,
+/// and a negative inset represents decreased distance from center.
+///
+/// # Examples
+///
+/// Positive insets produce larger rectangles:
```

Comment:
I would say "when added" just to be clear here, especially as I recommend implementing Sub also.